### ISSUE TITLE
rename to jni_gen ?

### ISSUE THREAD

**Issue by devoncarew** [dart-lang/jni_gen#3]
Hi - I'm not sure the intended lifetime of this package, but if we expect it to be around longer than the summer of code program, we should rename the repo and the package to `jni_gen` (from `jnigen`) to match our other repo names.

**Comment by devoncarew** [comment]
Gotcha, thanks for the background. However, we'll still want the Dart package to use standard pub naming conventions (https://dart.dev/tools/pub/pubspec#name - lowercase with underscores), and for the repo name to follow the package name.

**Comment by devoncarew** [comment]
Now that I look, I do see that ffigen is also not using pub conventions - https://github.com/dart-lang/ffigen/blob/master/pubspec.yaml#L5).

**Comment by devoncarew** [comment]
I agree that changing ffigen would be unnecessarily expensive. Changing jnigen would be cheap however, and avoid us creating new packages that don't follow pub conventions.

**Comment by dcharkes** [comment]
I've hit the rename button. 🚀  (Sorry for breaking your bookmarks/PRs/etc. 😆)

@mahesh-hegde could you make a PR to rename the stub-project? (Or fold it into the PR that moves it.)

**Comment by devoncarew** [comment]
👍 - thanks!

**Comment by dcharkes** [comment]
As noted by @mit-mit and @lrhn this causes our command line invocations to be `dart run jni_gen` which is less usable than `dart run jnigen`. Maybe we should reconsider this decision.

The standard pub naming conventions seem to be conflicting with usable commandline commands. @jonasfj @devoncarew

**Comment by mahesh-hegde** [comment]
I think that's not a frequently run command and jni_gen is also fine.

For comparison, the generator command for json_serializable is `dart run
build_runner build`.

        Mahesh

**Comment by mit-mit** [comment]
I worry especially about this due to mismatch between `ffigen` and `jni_gen` given these tools are likely to be used by the same users.

I think consistency between these is a LOT more important than consistency with other tools (like `build_runner`).

**Comment by mahesh-hegde** [comment]
Actually on top of this thread you can see me making the same argument.

But then @devoncarew's point also made sense.

Let's see any other arguments pros/cons come up in this discussion.

        Mahesh

**Comment by devoncarew** [comment]
I'm a little less concerned here than I was earlier; `jnigen` gives us consistently with `ffigen`, and, it seems like a more natural name for a tool (more natural than `jni_gen`).

**Comment by dcharkes** [comment]
@mahesh-hegde

When would be the best moment to change the repository name, again?
And when would be the best moment to change the `jni_gen` folder to `jnigen` again?
So that we can create the least amount of extra work. 😅

**Comment by mahesh-hegde** [comment]
After some features are implemented, at the end of GSoC time probably.

Continuing my track record of giving wildly inaccurate time estimates, that
would be around 3rd September. 😅

        Mahesh

**Comment by dcharkes** [comment]
Okay, we should just do it before we release a first version of the package on pub.dev.

**Comment by jonasfj** [comment]
> Okay, we should just do it before we release a first version of the package on pub.dev.

yes, `pub.dev` won't allow you to publish both `jnigen` and `jni_gen` because the names are too close.

**Comment by dcharkes** [comment]
https://github.com/dart-lang/jnigen/pull/47
